### PR TITLE
Normalizes project components table

### PR DESCRIPTION
**Pull request by tillyw**
## Associated issues
fixes https://github.com/cityofaustin/atd-data-tech/issues/10718

this pr normalizes the project components table by removing the `name` column, replaces any uses of that field in the Moped Editor with the associated `moped_proj_components` `component_name` column (including the `project_geography` view), and drops the `moped_proj_features` table which is no longer being used.

## Testing
**URL to test:** 

**Steps to test:**

- create a new project
- create a new component within that project
- edit an attribute and save
- delete the component
- create a new signal project
- repeat the steps above for the signal project

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

**Review by mddilley — COMMENTED** [review]
Noting that this depends on John's bug fix https://github.com/cityofaustin/atd-moped/pull/1025

Tested this locally with John's fix pulled in and all components add/edit/delete works great! Thanks for cleaning this up, and I'll turn this into an approval when the dependencies are sorted out! 🙏 🚀

File: moped-database/migrations/1681856452639_normalize_project_components_table/up.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        uniform_features.attributes AS attributes,
+        uniform_features.geography AS geography
+    FROM
+        moped_project
+        JOIN moped_proj_components ON (moped_proj_components.project_id = moped_project.project_id)
+        JOIN moped_components ON (moped_proj_components.component_id = moped_components.component_id)
+        JOIN feature_layers ON (moped_components.feature_layer_id = feature_layers.id)
+        JOIN uniform_features ON (moped_proj_components.project_component_id = uniform_features.component_id)
+        --join component_feature_map on (moped_proj_components.project_component_id = component_feature_map.component_id)
+        --join uniform_features on (component_feature_map.feature_id = uniform_features.id)
```
Comment (johnclary) [inline moped-database/migrations/1681856452639_normalize_project_components_table/up.sql:23]: i think it'd be best to remove these commented out parts (i see we overlooked them in previous migrations)

**Review by johnclary — APPROVED** [review]
@tillyw thanks so much for cleaning this up! love to see all the red ❌ 

i also tested this locally with the [component edit fix](https://github.com/cityofaustin/atd-moped/pull/1025 ) incorporated. 

🚢🚢🚢🚢

**Comment by johnclary** [comment]
@tillyw i'm merging for today's v1.22 release

**Pull request merged**